src/gnc/relative_orbit_estimate.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2020 Pathfinder for Autonomous Navigation (PAN)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+/** @file gnc_relative_orbit_estimator.cpp
+ *  @author Kyle Krol
+ */
+
+#include <gnc/config.hpp>
+#include <gnc/relative_orbit_estimate.hpp>
+#include <gnc/utilities.hpp>
+
+#include <lin/core.hpp>
+#include <lin/factorizations.hpp>
+#include <lin/generators.hpp>
+#include <lin/math.hpp>
+#include <lin/queries.hpp>
+#include <lin/references.hpp>
+#include <lin/substitutions.hpp>
+
+#include <tuple>
+
+namespace gnc {
+
+RelativeOrbitEstimate::Matrix<6, 6>
+RelativeOrbitEstimate::_state_transition_matrix(Time dt_ns, Real n) {
+  static constexpr Real zero  = 0.0;
+  static constexpr Real one   = 1.0;
+  static constexpr Real two   = 2.0;
+  static constexpr Real three = 3.0;
+  static constexpr Real four  = 4.0;
+  static constexpr Real six   = 6.0;
+
+  /* State transition matrix using the Clohessy Wiltshire equations:
+   *
+   *   F = [ Phi_rr(dt), Phi_rv(dt) ]
+   *       [ Phi_vr(dt), Phi_vv(dt) ].
+   *
+   * Reference(s):
+   *  - https://en.wikipedia.org/wiki/Clohessy%E2%80%93Wiltshire_equations
+   */
+  Real const dt  = static_cast<Real>(dt_ns) * Real(1.0e-9);
+  Real const nt  = n * dt;
+  Real const snt = lin::sin(nt);
+  Real const cnt = lin::cos(nt);
+
+  return {
+    four - three * cnt,    zero, zero,     snt / n,               two * (one - cnt) / n,         zero,
+    six * (snt - nt),      one,  zero,     two * (cnt - one) / n, (four * snt - three * nt) / n, zero,
+    zero,                  zero, cnt,      zero,                  zero,                          snt / n,
+    three * n * snt,       zero, zero,     cnt,                   two * snt,                     zero,
+    six * n * (cnt - one), zero, zero,     -two * snt,            four * cnt - three,            zero,
+    zero,                  zero, -n * snt, zero,                  zero,                          cnt
+  };
+}
```

Comment:
This function is incredibly important! Please verify the matrix elements against the linked reference. I've gone over them myself multiple times but we gotta make sure we get these right!